experimental/batchmap/ctmap/cmd/build/build.go

```diff
@@ -0,0 +1,223 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// build is a tool to build a map from a given clone of a log.
+package main
+
+import (
+	"bufio"
+	"context"
+	"flag"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/apache/beam/sdks/go/pkg/beam"
+	"github.com/apache/beam/sdks/go/pkg/beam/io/databaseio"
+	"github.com/apache/beam/sdks/go/pkg/beam/io/filesystem"
+	"github.com/apache/beam/sdks/go/pkg/beam/io/textio"
+	beamlog "github.com/apache/beam/sdks/go/pkg/beam/log"
+	"github.com/apache/beam/sdks/go/pkg/beam/x/beamx"
+	"github.com/golang/glog"
+	"github.com/google/trillian-examples/clone/logdb"
+	"github.com/google/trillian-examples/experimental/batchmap/ctmap/internal/pipeline"
+	"github.com/google/trillian/experimental/batchmap"
+
+	_ "github.com/go-sql-driver/mysql"
+)
+
+var (
+	mysqlURI         = flag.String("mysql_log_uri", "", "URL of a MySQL database to read the log from.")
+	mapOutputRootDir = flag.String("map_output_root_dir", "", "Filesystem root for this map. Cannot be shared with other maps.")
+	treeID           = flag.Int64("tree_id", 12345, "The ID of the tree. Used as a salt in hashing.")
+	prefixStrata     = flag.Int("prefix_strata", 2, "The number of strata of 8-bit strata before the final strata.")
+	count            = flag.Int64("count", 1, "The total number of entries starting from the beginning of the log to use")
+)
+
+func init() {
+	beam.RegisterType(reflect.TypeOf((*writeTileFn)(nil)).Elem())
+	beam.RegisterType(reflect.TypeOf((*writeCheckpointFn)(nil)).Elem())
+}
+
+func main() {
+	flag.Parse()
+	beam.Init()
+
+	ctx := context.Background()
+	if !strings.HasSuffix(*mapOutputRootDir, "/") {
```

Comment:
Might be worth a comment somewhere about the remote usecase - funnily enough I just wrote a draft comment about using `filepath` instead because someone might run this on windows :)